natlib/svcd.c

```diff
@@ -185,3 +186,41 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+static int svcd_unsubscribe( lua_State *L )
+{
+    // check number of args	
+    if (lua_gettop(L) != 4) return luaL_error(L, "Expected (targetip, svcid, attrid, ivkid)");
+    
+    uint16_t svcid = luaL_checknumber(L, 2);
+    uint16_t attrid = luaL_checknumber(L, 3);
+    uint16_t ivkid = luaL_checknumber(L, 4);
+    // set array to var msg
+    
+    lua_getglobal(L, "SVCD"); //index 5
+
+    //set oursubs of ivkid to nil
+    lua_pushstring(L, "oursubs");
+    lua_gettable(L, 5); 
+    lua_pushvalue(L, 2); //svc_id @ index ?
+    lua_pushnil(L); //set equal to nil?
+    lua_settable(L, 5);
+
+    //send msg over socket
+    lua_pushlightfunction(L, libstorm_net_sendto);
+    lua_pushstring(L, "ncsock");
+    lua_gettable(L, 5);
+    //msg as str
+    lua_pushlightfunction(L, arr_as_str);
+    uint8_t msg[7];
+    msg[0] = 0;
+    ((uint16_t*) (msg+1))[0] = svcid;
```

Comment:
Something is still fishy here. Firstly you are overwriting locations, and secondly you ignored my comment about unaligned writes. This is unsafe code.